include/lidar_processor/lidar_processor.hpp

```diff
@@ -38,18 +47,34 @@ class LidarProcessor : public rclcpp::Node
   explicit LidarProcessor(rclcpp::NodeOptions options);
 
 private:
+  float ground_point_model_threshold_{};
+  sensor_msgs::msg::Imu::SharedPtr last_imu_{};
+  sensor_msgs::msg::PointCloud2::SharedPtr last_pcl_{};
+  rclcpp::TimerBase::SharedPtr param_update_timer_;
+
+  void passthrough_stage(pcl::PointCloud<pcl::PointXYZI>::Ptr cloud);
+  void ground_segmentation(pcl::PointCloud<pcl::PointXYZI>::Ptr cloud, pcl::PointCloud<pcl::PointXYZI>::Ptr ground);
+
+  void update_params();
   void raw_ls_callback(const sensor_msgs::msg::LaserScan::SharedPtr msg);
   void raw_pc_callback(const sensor_msgs::msg::PointCloud2::SharedPtr msg);
+  void imu_callback(const sensor_msgs::msg::Imu::SharedPtr msg);
```

Comment:
instead of adding the IMU callback to the Lidarprocessor class, we should move it to its own class to improve maintainability of the code

Reply:
nvm, didn't see that ground segmentation was using it

Reply:
its needed so that we can filter ground points. im not sure i agree with you here. how would you like to see it?

Reply:
yeah I know, I didnt see that at the time of posting that comment. sorry!!

Reply:
nw, thanks for looking it over!